<launch>

    <master auto="start"/>
    <remap from="/cmd_vel" to="/carla/ego_vehicle/twist_pid"/>

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find psaf_steering)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find psaf_steering)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find psaf_steering)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find psaf_steering)/config/global_costmap_params.yaml" command="load" />

        <param name="base_local_planner" value="psaf_local_planner/PsafLocalPlanner"/>
        <param name="base_global_planner" value="psaf_global_planner/GlobalPlanner" />
        <param name="controller_frequency" value="20.0" />
    </node>

    <include file="$(find psaf_steering)/launch/carla_twist_pid_control_node.launch" />

    <include file="$(find psaf_sensor_preprocessing)/launch/psaf_semantic_processing.launch"/>

</launch> 